// react-native-month/lib/commonjs/components/WeekDays/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const SHOULD_NOT_UPDATE = true;

const WeekColumn = /*#__PURE__*/_react.default.memo(props => /*#__PURE__*/_react.default.createElement(_reactNative.View, {
  style: [{
    flex: 1,
    alignItems: 'center'
  }, props.theme.weekColumnStyle]
}, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
  allowFontScaling: false,
  style: props.theme.weekColumnTextStyle
}, props.day)), () => SHOULD_NOT_UPDATE);

var _default = /*#__PURE__*/_react.default.memo(props => /*#__PURE__*/_react.default.createElement(_reactNative.View, {
  style: [{
    flexDirection: 'row'
  }, props.theme.weekColumnsContainerStyle]
}, props.days.map(day => /*#__PURE__*/_react.default.createElement(WeekColumn, {
  key: day,
  day: day,
  theme: props.theme
}))), () => SHOULD_NOT_UPDATE);

exports.default = _default;
