// react-native-month/lib/commonjs/components/Month/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _date = require("../../utils/date");

var _utils = require("../utils");

var _WeekDays = _interopRequireDefault(require("../WeekDays"));

var _Day = _interopRequireDefault(require("../Day"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = /*#__PURE__*/_react.default.memo(props => {
  const {
    month,
    year,
    onPress,
    locale = 'en',
    dayNames,
    showWeekdays,
    disabledDays = {},
    disableRange = false,
    disableOffsetDays = false,
    firstDayMonday = true,
    startDate,
    endDate,
    minDate,
    maxDate,
    markedDays = {},
    theme = {},
    renderDayContent
  } = props;
  const DAY_NAMES = Array.isArray(dayNames) && dayNames.length === 7 ? dayNames : (0, _date.getDayNames)(locale, firstDayMonday);
  const days = (0, _utils.getMonthDays)(month, year, firstDayMonday, disableRange, disabledDays, disableOffsetDays, startDate, endDate, minDate, maxDate);
  const weeks = [];

  while (days.length) {
    weeks.push(days.splice(0, 7));
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, showWeekdays && /*#__PURE__*/_react.default.createElement(_WeekDays.default, {
    days: DAY_NAMES,
    theme: theme
  }), weeks.map((week, index) => /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    key: String(index),
    style: {
      flexDirection: 'row'
    }
  }, week.map(day => /*#__PURE__*/_react.default.createElement(_Day.default, {
    key: day.key,
    item: day,
    dots: markedDays[day.id] ? markedDays[day.id].dots : [],
    onPress: onPress,
    theme: theme,
    renderDayContent: renderDayContent
  })))));
}, _utils.areEqual);

exports.default = _default;
